### PR TITLE
Js add typing

### PR DESCRIPTION
**Pull request by heeler**
I added type hints and created an exception rather than using the assert mechanism.

### CODE REVIEW

File: geneselection/datasets/dataset.py [hunk 1]
```diff
@@ -2,55 +2,75 @@
...
+        uns: dict={}
     ):
         """
         A data provider class for the larger project. The idea is to capture an AnnData and then
         serve up the rows as either pytorch Tensors, or numpy ndarrays based on the keyword argument
-        :param data: An AnnData object which is copied to protect from changes to the original object.
+        :param X: a Tensor created from for example an (ndarray)AnnData.X data block
+        :param obs: a pandas.DataFrame containing the observational labels corresponding to AnnData.obs
+        :param var: a pandas.DataFrame containing the variable labels contained in an AnnData.obs
+        :param uns: a pandas.DataFrame containing the unstructured descriptors contained in AnnData.uns
```
Comment (donovanr) [inline geneselection/datasets/dataset.py]: this should be a dict
```suggestion
        :param uns: a dict containing the unstructured descriptors contained in AnnData.uns
```

File: geneselection/datasets/dataset.py [hunk 2]
```diff
@@ -2,55 +2,75 @@
...
+class GSDatasetVarMismatchError(Exception):
+    """Exception for when asked to merge 2 GSDatasets with different var classes (columns)"""
+    pass
 
 
 class GSDataset(Dataset):
     def __init__(
-        self, X=torch.zeros(1, 1), obs=pd.DataFrame([0]), var=pd.DataFrame([0]), uns={}
+        self,
+        x: Tensor=torch.zeros(1, 1),
```
Comment (donovanr) [inline geneselection/datasets/dataset.py]: why did  X -> x here?

File: geneselection/datasets/dataset.py [hunk 3]
```diff
@@ -2,55 +2,75 @@
...
+        self.X = x
         self.obs = obs
         self.var = var
         self.uns = uns
 
     def __len__(self) -> int:
         return len(self.X)
 
-    def _get_item(self, idx):
+    def _get_item(self, idx: int) -> Dict[Tensor, pd.Series]:
```
Comment (donovanr) [inline geneselection/datasets/dataset.py]: i think this should be Dict[Tensor, pd.DataFrame], not Series

File: geneselection/datasets/dataset.py (hunk 2 above)
Comment (heeler) [reply geneselection/datasets/dataset.py]: I know it's stupid but the Pep standard complains about uppercase variables. To me, it makes sense Uppercase for matrix but having the IDE highlight it was annoying.

File: geneselection/datasets/dataset.py (hunk 1 above)
Comment (heeler) [reply geneselection/datasets/dataset.py]: fixed in commit [dac9679](https://github.com/AllenCellModeling/geneselection/commit/dac9679f5fe935554321300b5af50e25cf0a3406)

**Pull request merged**